src/helpers/ethereumAuthenticate.js

```diff
@@ -0,0 +1,34 @@
+const ethers = require('ethers')
+
+async function authenticateFromConfig(ethereumConfig, log = console.info) {
+    let wallet
+    let provider
+
+    if (ethereumConfig.url) {
+        provider = new ethers.providers.JsonRpcProvider(ethereumConfig.url)
+        // eslint-disable-next-line no-underscore-dangle
+        await provider._networkPromise
+    }
+    if (ethereumConfig.privateKey) {
+        log('Ethereum Authentication with private key')
+        try {
+            wallet = await new ethers.Wallet(ethereumConfig.privateKey, provider)
+        } catch (e) {
+            throw new Error(e)
+        }
+    } else if (ethereumConfig.generateWallet) {
+        log('Ethereum authentication with new randomly generated wallet')
+        try {
+            wallet = await ethers.Wallet.createRandom()
```

Comment:
`try {} catch {}` is redundant here. Unless intention is to highlight that an exception could be thrown.